modules/lead/product-base/database.tf

```diff
@@ -36,5 +32,12 @@ resource "helm_release" "mongodb" {
   timeout    = 600
   wait       = true
 
-  values = [data.template_file.mongo_values.rendered]
-}
\ No newline at end of file
+  values = [
+    templatefile("${path.module}/mongo.tpl", {
+      mongodbRootPassword = random_password.mongodb_root_password.result
+    })
+  ]
+  depends_on = [
+    random_password.mongodb_root_password
```

Comment:
nitpick: this is not needed since you are already referencing this resource on line 37 which creates an implicit dependency.

I don't care if you remove it or not, just wanted to let you know.

Reply:
Good eye here, I initially added this to make _sure_ the resource was there for the helm chart installation but as long as Terraform knows to wait I'm fine with removing it.